src/lib/core/a11y/interactivity-checker.ts

```diff
@@ -30,26 +33,93 @@ export class InteractivityChecker {
    * being clipped by an `overflow: hidden` parent or being outside the viewport.
    */
   isVisible(element: HTMLElement) {
-    // There are additional special cases that this does not capture, but this will work for
-    // the most common cases.
+    let nodeName = element.nodeName.toLowerCase();
+    let isControlAudio = nodeName === 'audio' && (element as HTMLAudioElement).controls;
 
-    // Use logic from jQuery to check for `display: none`.
-    // See https://github.com/jquery/jquery/blob/master/src/css/hiddenVisibleSelectors.js#L12
-    if (!(element.offsetWidth || element.offsetHeight || element.getClientRects().length)) {
+    // In IE11 audio elements with controls have invalid rectangles, but are still visible.
+    if (!isControlAudio && !checkRectangles()) {
       return false;
     }
 
-    // Check for css `visibility` property.
-    // TODO(jelbourn): do any browsers we support return an empty string instead of 'visible'?
-    return getComputedStyle(element).getPropertyValue('visibility') == 'visible';
+    return getComputedStyle(element).getPropertyValue('visibility') === 'visible';
+
+    function checkRectangles() {
```

Comment:
This should be a module method with a description, similar to the others at the bottom of this file. I'd also call it something like `hasGeometry` or `elementHasRenderedRect`